### PR TITLE
add --fallbackmode to handily set a mode to files, due to metadata un…

### PR DESCRIPTION
**Pull request by aligo**
…supported by skd

or hope to support metadata early

### CODE REVIEW

**Comment by frostyplanet** [comment]
what do you mean by "skd" ?

**Comment by aligo** [comment]
@frostyplanet here it comes: https://github.com/yunify/qsfs-fuse/blob/7fd39a74499712a2ed18b7a6ac2f377fa4763c9a/src/client/QSClientConverter.cpp#L158

**Comment by aligo** [comment]
In my case, want to mount qsfs in fstab, which would made qsfs process run as root, meanwhile my app runs as another user, can't have a proper access to files defined mode 0644 and uid/gid both mismatched.

**Comment by frostyplanet** [comment]
skd actually a typo of SDK ...

File: src/filesystem/Parser.cpp [hunk 1]
```diff
@@ -107,6 +107,8 @@ static struct options {
   bool allowOther;
 
   int qsfsMultiThread;       // internal use only to turn on multi thread
+
+  mode_t fallbackMode;     // default fallback mode
```
Comment (jimhuaang) [inline src/filesystem/Parser.cpp]: What does fallback means here? Seems "fallbackMode" is not a great name. what we are doing here is to provide a option to specifying a file mode (OCTAL-MODE format), I think we could just name it fileMode or some other name which can clearly show the intention.

Comment (jimhuaang) [inline src/filesystem/Parser.cpp]: And could you also update the docs in HelpText.cpp::ShowQSFSHelp and HelpText.cpp::ShowQSFSUsage?

Comment (jimhuaang) [reply src/filesystem/Parser.cpp]: I think this option is relating to the stat, we better group this new option with other stat releated options (maxstat and statexpire). I mean we better to add this new option following statexpire, this also applies to your changes in other files.

**Review by jimhuaang — CHANGES_REQUESTED** [review]
Looks good. I have a few comments that should be addressed before this request get merged.

**Comment by aligo** [comment]
@jimhuaang fallbackmode => filemode, Put it above network related options.

File: src/filesystem/HelpText.cpp [hunk 2]
```diff
@@ -75,6 +78,8 @@ void ShowQSFSHelp() {
   "  -L, --loglevel     Min log level, message lower than this level don't logged;\n"
   "                     Specify one of following log level: INFO,WARN,ERROR,FATAL;\n"
   "                     " << GetDefaultLogLevelName() << " is set by default\n"
+  "  -F, --filemode     Specify a file mode for all without mode in metadata, default value\n"
+  "                     is " << ModeToString(QS::Configure::Options::Instance().GetFileMode()) << " times\n"
```
Comment (jimhuaang) [inline src/filesystem/HelpText.cpp]: there is a typo, " times" is not needed here

**Review by jimhuaang — CHANGES_REQUESTED** [review]
Please see the comments, there is a typo

**Review by jimhuaang — CHANGES_REQUESTED** [review]
Please also replace GetDefineDirMode, GetDefineFileMode used  in other places (FileMetaData.cpp, Operations.cpp) with your new code (GetFileMode).

**Review by jimhuaang — CHANGES_REQUESTED** [review]
Utils.cpp still have one GetDefineDirMode need to be replaced. thanks.

**Comment by aligo** [comment]
That one looks like used to create local directory, keep that way would be ok.

**Review by jimhuaang — APPROVED** [review]

**Pull request merged**